File: matlab/distance_btw_point_line.m
function [x,y,distance] = distance_btw_point_line(point, p1, p2)
%
%[x,y,distance] = distance_point_btw_line([0,0],[0,-4],[-4,0])
%Results:
%x = -2
%y = -2
%distance =  2.8284

    x1 = p1(1); y1 = p1(2);
    x2 = p2(1); y2 = p2(2);
    xp = point(1); yp = point(2);
    
    Mo = (y2-y1)/(x2-x1); %(slope of line between p1 to p2)
    Mn = -Mo;
    
    K1=Mo*x1-y1;
    
    K2=Mn*xp-yp;
    
    x = (K1-K2)/(2*Mo);
    
    y = Mo*(-K1-K2)/(2*Mo);
    
    distance = sqrt((xp-x)^2+(yp-y)^2);    
end 

File: cpp/distance_btw_point_line.cpp
// Example program
#include <iostream>
#include <math.h>

struct Point
{
  double x;
  double y;
};

double distance_btw_point_line(Point po, Point p1, Point p2){
    double x, y, Mo, Mn, K1, K2;
    double distance;
    
    Mo = (p2.y-p1.y)/(p2.x-p1.x); //slope of line between p1 to p2
    Mn = -Mo;
    
    K1=Mo*p1.x-p1.y;
    
    K2=Mn*po.x-po.y;
    
    x = (K1-K2)/(2*Mo);
    
    y = (-K1-K2)/2;
    
    distance = sqrt((po.x-x)*(po.x-x)+(po.y-y)*(po.y-y));
    
    return distance;
}
Point hit_point_line(Point po, Point p1, Point p2){
    double Mo, Mn, K1, K2;
    Point p;

    Mo = (p2.y-p1.y)/(p2.x-p1.x); //slope of line between p1 to p2
    Mn = -Mo;
    
    K1=Mo*p1.x-p1.y;
    
    K2=Mn*po.x-po.y;
    
    p.x = (K1-K2)/(2*Mo);
    
    p.y = (-K1-K2)/2;
    
    return p;
}

int main()
{
  Point A = {0,0};
  Point B1 = {4,0};
  Point B2 = {0,4};
  double dist = distance_btw_point_line(A,B1,B2);
  Point hp = hit_point_line(A,B1,B2);
  printf("Hit Point = (%2.2f , %2.2f) \n",hp.x,hp.y);
  printf("Distance: %2.2f",dist);
}
